tests-functional/tests/test_cases.py

```diff
@@ -150,3 +151,65 @@ def setup_method(self):
         websocket_thread = threading.Thread(target=self.signal_client._connect)
         websocket_thread.daemon = True
         websocket_thread.start()
+
+
+class NetworkConditionTestCase:
+
+    @contextmanager
+    def add_latency(self):
+        pass
+        #TODO: To be implemented when we have docker exec capability
+
+    @contextmanager
+    def add_packet_loss(self):
+        pass
+        #TODO: To be implemented when we have docker exec capability
+
+    @contextmanager
+    def add_low_bandwith(self):
+        pass
+        #TODO: To be implemented when we have docker exec capability
+
+    @contextmanager
+    def node_pause(self, node):
+        pass
+        #TODO: To be implemented when we have docker exec capability
+
+class OneToOneMessageTestCase(NetworkConditionTestCase):
+
+    def initialize_backend(self, await_signals, display_name=DEFAULT_DISPLAY_NAME, url=None):
+        backend = StatusBackend(await_signals=await_signals, url=url)
+        backend.init_status_backend()
+        backend.create_account_and_login(display_name=display_name)
+        backend.start_messenger()
+        return backend
+
+
+    def validate_event_against_response(self, event, fields_to_validate, response):
+        messages_in_event = event["event"]["messages"]
+        assert len(messages_in_event) > 0, "No messages found in the event"
+        response_chat = response["result"]["chats"][0]
+
+        mismatch_details = []
+        for message in messages_in_event:
+            match = True
+            message_mismatch = []
+            for response_field, event_field in fields_to_validate.items():
+                response_value = response_chat["lastMessage"][response_field]
+                event_value = message[event_field]
+                if response_value != event_value:
+                    match = False
+                    message_mismatch.append(
+                        f"Field '{response_field}': Expected '{response_value}', Found '{event_value}'"
+                    )
+            if match:
+                return
```

Comment:
Perhaps continue the outer loop here? 🤔 
```suggestion
                continue
```

Reply:
Good question! 
We return because we want at least one match the message we sent in this particular `wakuext_sendOneToOneMessage` request. 
From what I saw there can be multiple messages in the signal but only one is the message that we care about. If we continue the tests will fail with mismatches.

<details>
    <summary>Here's an example of a signal that contains 3 messages</summary>

```json
{
    "event": {
        "chats": [
            {
                "active": false,
                "alias": "zQ3...ZwyVps",
                "chatType": 1,
                "color": "",
                "deletedAtClockValue": 0,
                "description": "",
                "emoji": "",
                "highlight": true,
                "id": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "identicon": "",
                "joined": 1732639227332,
                "lastClockValue": 1732639227534,
                "lastMessage": {
                    "alias": "zQ3...ZwyVps",
                    "chatId": "0x04bbfad6c3d7dcfc2eecc525cbda889d66433a745e551f7cec1cb499e61dba46bf5f51db2172b7b290fa10f584623486f29a1a9281210b62a3dc7a4951e87b547d",
                    "clock": 1732639227534,
                    "compressedKey": "zQ3shss5nftQmQpoJdzn4vhZtenH8X1pyaXWPUWaxLNZwyVps",
                    "contentType": 1,
                    "displayName": "Mr_Meeseeks",
                    "emojiHash": [
                        "🧑🏻‍⚖️",
                        "🌳",
                        "🦴",
                        "👊🏿",
                        "🧝🏼‍♀️",
                        "💁🏻‍♂️",
                        "🦘",
                        "👮🏾‍♀️",
                        "👩🏼‍🦼",
                        "🧽",
                        "🥝",
                        "🪐",
                        "👩🏿‍🔬",
                        "🍯"
                    ],
                    "ensName": "",
                    "from": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                    "id": "0x3b75338b4894b6b5a1773188d9ea63170fae7dd29136bd6deb9cddd54e719c60",
                    "identicon": "",
                    "lineCount": 0,
                    "localChatId": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                    "messageType": 1,
                    "new": true,
                    "parsedText": [
                        {
                            "children": [
                                {
                                    "literal": "test_message_2_92d72c14-1579-46e5-9aec-b4b397953ee6"
                                }
                            ],
                            "type": "paragraph"
                        }
                    ],
                    "quotedMessage": null,
                    "replace": "",
                    "responseTo": "",
                    "rtl": false,
                    "seen": false,
                    "text": "test_message_2_92d72c14-1579-46e5-9aec-b4b397953ee6",
                    "timestamp": 1732639227534,
                    "whisperTimestamp": 1732639227000
                },
                "members": null,
                "membershipUpdateEvents": null,
                "muted": false,
                "muteTill": "0001-01-01T00:00:00Z",
                "name": "0x04c47a",
                "ReadMessagesAtClockValue": 0,
                "timestamp": 1732639239337,
                "unviewedMentionsCount": 5,
                "unviewedMessagesCount": 5,
                "viewersCanPostReactions": false
            }
        ],
        "contacts": [
            {
                "active": false,
                "added": false,
                "address": "0x2e0fbffcc0d97c8a0d043d8c7278bddbcb1fb164",
                "alias": "zQ3...ZwyVps",
                "bio": "",
                "blocked": false,
                "compressedKey": "zQ3shss5nftQmQpoJdzn4vhZtenH8X1pyaXWPUWaxLNZwyVps",
                "contactRequestLocalClock": 0,
                "contactRequestLocalState": 0,
                "contactRequestRemoteClock": 1732639227507,
                "contactRequestRemoteState": 3,
                "contactRequestState": 3,
                "customizationColor": "primary",
                "displayName": "Mr_Meeseeks",
                "emojiHash": [
                    "🧑🏻‍⚖️",
                    "🌳",
                    "🦴",
                    "👊🏿",
                    "🧝🏼‍♀️",
                    "💁🏻‍♂️",
                    "🦘",
                    "👮🏾‍♀️",
                    "👩🏼‍🦼",
                    "🧽",
                    "🥝",
                    "🪐",
                    "👩🏿‍🔬",
                    "🍯"
                ],
                "ensVerified": false,
                "hasAddedUs": true,
                "id": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "identicon": "",
                "images": null,
                "IsSyncing": false,
                "lastUpdated": 0,
                "lastUpdatedLocally": 0,
                "mutual": false,
                "primaryName": "Mr_Meeseeks",
                "Removed": false,
                "trustStatus": 0,
                "verificationStatus": 0
            }
        ],
        "discordOldestMessageTimestamp": 0,
        "messages": [
            {
                "alias": "zQ3...ZwyVps",
                "chatId": "0x04bbfad6c3d7dcfc2eecc525cbda889d66433a745e551f7cec1cb499e61dba46bf5f51db2172b7b290fa10f584623486f29a1a9281210b62a3dc7a4951e87b547d",
                "clock": 1732639227516,
                "compressedKey": "zQ3shss5nftQmQpoJdzn4vhZtenH8X1pyaXWPUWaxLNZwyVps",
                "contentType": 1,
                "displayName": "",
                "emojiHash": [
                    "🧑🏻‍⚖️",
                    "🌳",
                    "🦴",
                    "👊🏿",
                    "🧝🏼‍♀️",
                    "💁🏻‍♂️",
                    "🦘",
                    "👮🏾‍♀️",
                    "👩🏼‍🦼",
                    "🧽",
                    "🥝",
                    "🪐",
                    "👩🏿‍🔬",
                    "🍯"
                ],
                "ensName": "",
                "from": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "gapParameters": {
                },
                "id": "0x750831bb6eb1993ba1f72efa70f21f30a25fb040da3068eba92a2b81436ab4af",
                "identicon": "",
                "lineCount": 0,
                "localChatId": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "messageType": 1,
                "new": true,
                "parsedText": [
                    {
                        "children": [
                            {
                                "literal": "test_message_1_90f5c755-54ad-442f-88e4-f15160dfe9fb"
                            }
                        ],
                        "type": "paragraph"
                    }
                ],
                "quotedMessage": null,
                "replace": "",
                "responseTo": "",
                "rtl": false,
                "seen": false,
                "text": "test_message_1_90f5c755-54ad-442f-88e4-f15160dfe9fb",
                "timestamp": 1732639227516,
                "whisperTimestamp": 1732639227000
            },
            {
                "alias": "zQ3...ZwyVps",
                "chatId": "0x04bbfad6c3d7dcfc2eecc525cbda889d66433a745e551f7cec1cb499e61dba46bf5f51db2172b7b290fa10f584623486f29a1a9281210b62a3dc7a4951e87b547d",
                "clock": 1732639227507,
                "compressedKey": "zQ3shss5nftQmQpoJdzn4vhZtenH8X1pyaXWPUWaxLNZwyVps",
                "contactRequestState": 1,
                "contentType": 11,
                "displayName": "",
                "emojiHash": [
                    "🧑🏻‍⚖️",
                    "🌳",
                    "🦴",
                    "👊🏿",
                    "🧝🏼‍♀️",
                    "💁🏻‍♂️",
                    "🦘",
                    "👮🏾‍♀️",
                    "👩🏼‍🦼",
                    "🧽",
                    "🥝",
                    "🪐",
                    "👩🏿‍🔬",
                    "🍯"
                ],
                "ensName": "",
                "from": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "gapParameters": {
                },
                "id": "0xe379879b927f75d763dd392e1a87ff94ab811b98659cee312bf4306e3ebd14c3",
                "identicon": "",
                "lineCount": 0,
                "localChatId": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "messageType": 1,
                "new": true,
                "parsedText": [
                    {
                        "children": [
                            {
                                "literal": "contact_request"
                            }
                        ],
                        "type": "paragraph"
                    }
                ],
                "quotedMessage": null,
                "replace": "",
                "responseTo": "",
                "rtl": false,
                "seen": false,
                "text": "contact_request",
                "timestamp": 1732639227507,
                "whisperTimestamp": 1732639227000
            },
            {
                "alias": "zQ3...ZwyVps",
                "chatId": "0x04bbfad6c3d7dcfc2eecc525cbda889d66433a745e551f7cec1cb499e61dba46bf5f51db2172b7b290fa10f584623486f29a1a9281210b62a3dc7a4951e87b547d",
                "clock": 1732639227534,
                "compressedKey": "zQ3shss5nftQmQpoJdzn4vhZtenH8X1pyaXWPUWaxLNZwyVps",
                "contentType": 1,
                "displayName": "",
                "emojiHash": [
                    "🧑🏻‍⚖️",
                    "🌳",
                    "🦴",
                    "👊🏿",
                    "🧝🏼‍♀️",
                    "💁🏻‍♂️",
                    "🦘",
                    "👮🏾‍♀️",
                    "👩🏼‍🦼",
                    "🧽",
                    "🥝",
                    "🪐",
                    "👩🏿‍🔬",
                    "🍯"
                ],
                "ensName": "",
                "from": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "gapParameters": {
                },
                "id": "0x3b75338b4894b6b5a1773188d9ea63170fae7dd29136bd6deb9cddd54e719c60",
                "identicon": "",
                "lineCount": 0,
                "localChatId": "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5",
                "messageType": 1,
                "new": true,
                "parsedText": [
                    {
                        "children": [
                            {
                                "literal": "test_message_2_92d72c14-1579-46e5-9aec-b4b397953ee6"
                            }
                        ],
                        "type": "paragraph"
                    }
                ],
                "quotedMessage": null,
                "replace": "",
                "responseTo": "",
                "rtl": false,
                "seen": false,
                "text": "test_message_2_92d72c14-1579-46e5-9aec-b4b397953ee6",
                "timestamp": 1732639227534,
                "whisperTimestamp": 1732639227000
            }
        ],
        "notifications": null,
        "updatedProfileShowcaseContactIDs": [
            "0x04c47af76aa5b3ebc6807b84540fb2e1e43d19dff30fdb338e4acaca81265ad0a88ac22ce0ddc7807a7c8006c99217187f182750ba8c280453b0f357536307f0e5"
        ]
    },
    "type": "messages.new"
}
```
</details>

Reply:
@fbarbu15 Oh ok, I got the code now. This makes sense, thank you!
And sorry for the delay.

Perhaps the code could be improved to make the intentions clearer.
1. Find the message with according `id` in `messages_in_event`
2. Ensure all `fields_to_validate` match the found `message` (`id` doesn't need to be in `fields_to_validate` anymore).

```python
def validate_event_against_response(self, event, fields_to_validate, response):
    messages_in_event = event["event"]["messages"]
    assert len(messages_in_event) > 0, "No messages found in the event"
    response_chat = response["result"]["chats"][0]

    message_id = response_chat["lastMessage"]["id"]
    message = next((message for message in messages_in_event if message["id"] == message_id), None)
    assert message, f"Message with ID {message_id} not found in the event"

    message_mismatch = []
    for response_field, event_field in fields_to_validate.items():
        response_value = response_chat["lastMessage"][response_field]
        event_value = message[event_field]
        if response_value != event_value:
            message_mismatch.append(
                f"Field '{response_field}': Expected '{response_value}', Found '{event_value}'"
            )

    if not message_mismatch:
        return

    raise AssertionError(
        "Some Sender RPC responses are not matching the signals received by the receiver.\n"
        "Details of mismatches:\n" +
        "\n".join(message_mismatch)
    )
```

Reply:
thanks, indeed it makes more sense to match the message on ID first and then do the other checks.
Applied this change